pkg/k8s/apis/cilium.io/v2alpha1/vtep_policy_types.go

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium,ciliumpolicy},singular="ciliumvteppolicy",path="ciliumvteppolicies",scope="Cluster",shortName={vtep-policy}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+// +kubebuilder:storageversion
+
+type CiliumVtepPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec CiliumVtepPolicySpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumVtepPolicyList is a list of CiliumVtepPolicy objects.
+type CiliumVtepPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumVtepPolicy.
+	Items []CiliumVtepPolicy `json:"items"`
+}
+
+// +kubebuilder:validation:Format=cidr
+type CIDR string
+
+// +kubebuilder:validation:Pattern=`^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$`
+// regex source: https://uibakery.io/regex-library/mac-address
+type MAC string
+
+type CiliumVtepPolicySpec struct {
+	// +kubebuilder:validation:MaxItems=30
+	// CiliumVtepPolicyRules represents a list of rules by which traffic is
+	// selected from/to the pods.
+	Selectors []CiliumVtepPolicyRules `json:"selectors,omitempty"`
+
+	// +kubebuilder:validation:MaxItems=30
+	// DestinationCIDRs is a list of destination CIDRs for destination IP addresses.
+	// If a destination IP matches any one CIDR, it will be selected.
+	DestinationCIDRs []CIDR `json:"destinationCIDRs,omitempty"`
```

Comment:
What happens if multiple policies have the same selector + destination? They will likely to generate a same key for the vteppolicy map and the write will conflict at the dataplane level.